okhttp/src/main/java/okhttp3/internal/http2/Hpack.java

```diff
@@ -510,8 +512,16 @@ void writeInt(int value, int prefixMask, int bits) {
     }
 
     void writeByteString(ByteString data) throws IOException {
-      writeInt(data.size(), PREFIX_7_BITS, 0);
-      out.write(data);
+      if (useCompression && Huffman.get().encodedLength(data) < data.size()) {
```

Comment:
I’m curious about the runtime cost of calling `encodedLength()` and then `encode()`. . .

It might be more efficient in aggregate to always do a full encode and to just measure the buffer’s length. That’s going to be faster in the fast, common case when Huffman is useful, right?